crypto/finite-field.ts

```diff
@@ -300,10 +300,10 @@ function createField(
       return mod(x * x, p);
     },
     isSquare(x: bigint) {
-      return isSquare(x, p);
+      return isSquare(mod(x, p), p);
     },
     sqrt(x: bigint) {
-      return sqrt(x, p, oddFactor, twoadicRoot, twoadicity);
+      return sqrt(mod(x, p), p, oddFactor, twoadicRoot, twoadicity);
```

Comment:
so what I meant in my other review here is that I'd like those changes to be made to the original `sqrt` and `isSquare` functions defined at the _top-level_ of this file. just o that we don't have two versions of `sqrt()`, one with a bug and another that fixes the bug

Reply:
I know that currently the original `sqrt()` / `isSquare()` are not even exported directly, but other basic functions are, and if someone decides to export `sqrt()` directly in the future they probably won't think of fixing its bug

Reply:
fixed here! https://github.com/o1-labs/o1js-bindings/pull/287/commits/cc3f63228f6caa025a39b761b1e38faa8c50e69e